test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -643,3 +643,80 @@ Describe "Invoke-RestMethod tests" -Tags "Feature" {
         $result.Error | Should BeNullOrEmpty
     }
 }
+
+Describe "Bug fixes" -Tags "Feature" {
+
+    Context "InFile parameter negative tests" {
+
+        $testCases = @(
+            @{
+                Name = 'Validate error for Invoke-WebRequest -InFile ""'
+                ScriptBlock = {Invoke-WebRequest -Uri http://httpbin.org/post -Method Post -InFile ""}
+                ExpectedFullyQualifiedErrorId = 'WebCmdletInFileNotFilePathException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand'
+            }
+
+            @{
+                Name = 'Validate error for Invoke-WebRequest -InFile'
+                ScriptBlock = {Invoke-WebRequest -Uri http://httpbin.org/post -Method Post -InFile}
+                ExpectedFullyQualifiedErrorId = 'MissingArgument,Microsoft.PowerShell.Commands.InvokeWebRequestCommand'
+            }
+
+            @{
+                Name = "Validate error for Invoke-WebRequest -InFile  $TestDrive\content.txt"
+                ScriptBlock = {Invoke-WebRequest -Uri http://httpbin.org/post -Method Post -InFile  $TestDrive\content.txt}
+                ExpectedFullyQualifiedErrorId = 'PathNotFound,Microsoft.PowerShell.Commands.InvokeWebRequestCommand'
+            }
+
+            @{
+                Name = "Validate error for Invoke-RestMethod -InFile ''"
+                ScriptBlock = {Invoke-RestMethod -Uri http://httpbin.org/post -Method Post -InFile ''}
+                ExpectedFullyQualifiedErrorId = 'WebCmdletInFileNotFilePathException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand'
+            }
+
+            @{
+                Name = "Validate error for Invoke-RestMethod -InFile <null>"
+                ScriptBlock = {Invoke-RestMethod -Uri http://httpbin.org/post -Method Post -InFile}
+                ExpectedFullyQualifiedErrorId = 'MissingArgument,Microsoft.PowerShell.Commands.InvokeRestMethodCommand'
+            }
+
+            @{
+                Name = "Validate error for Invoke-RestMethod -InFile  $TestDrive\content.txt"
+                ScriptBlock = {Invoke-RestMethod -Uri http://httpbin.org/post -Method Post -InFile $TestDrive\content.txt}
+                ExpectedFullyQualifiedErrorId = 'PathNotFound,Microsoft.PowerShell.Commands.InvokeRestMethodCommand'
+            }
+        )
+
+        It "<Name>" -TestCases $testCases {
+            param ($scriptblock, $expectedFullyQualifiedErrorId)
+
+            try
+            {
+                & $scriptblock
+                throw "CodeExecuted"
+            }
+            catch
+            {
+                $_.FullyQualifiedErrorId | should be $ExpectedFullyQualifiedErrorId
+            }
+        }
+    }
+
+    Context "InFile parameter positive tests" {
```

Comment:
Yes, I agree. For Full PS, we had a small web service that validate some basic functionality. We have to options:
1) Create our own web service
2) host our own version of  http://httpbin.org (See issue #3012)